Source/Core/DolphinQt/CMakeLists.txt

```diff
@@ -152,11 +155,71 @@ PRIVATE
 )
 
 if(WIN32)
-  target_sources(dolphin-emu PRIVATE DolphinQt.manifest)
+  target_sources(dolphin-emu PRIVATE DolphinQt.manifest DolphinQt.rc)
+
+  if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(Dolphin_NAME "DolphinD")
+  else()
+    set(Dolphin_NAME "Dolphin")
+  endif()
+
+  set_target_properties(dolphin-emu PROPERTIES
+    WIN32_EXECUTABLE TRUE
+    OUTPUT_NAME ${Dolphin_NAME}
+  )
+
+  target_compile_options(dolphin-emu PRIVATE /D _SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING)
```

Comment:
`target_compile_definitions(dolphin-emu PRIVATE "-D_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING")`

Otherwise, this adds a rogue /D to the resource compiler command, causing it to write DolphinQt.rc.res to the source directory instead of the build directory.